one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -114,6 +121,23 @@ public void onError(String commandName, Exception exception) {
         return payload;
     }
 
+    @RequestMapping(value = "/project-download", produces = "application/zip")
+    public void downloadProject(HttpServletRequest request, HttpServletResponse response) {
+        HubProject project = hubConfig.getHubProject();
+        File exportFile = project.getProjectDir().resolve("build").resolve("datahub-project.zip").toFile();
+        project.exportProject(exportFile);
```

Comment:
We can get this done in one write - instead of calling this method, add an overloaded method to HubProject - e.g. exportProject(OutputStream) - and then pass in response.getOutputStream. That method will wrap a ZipOutputStream around it and write the contents directly to the HTTP response. That avoids messing around with files.